test/test-framework/pkg/apis/cache/v1alpha1/memcached_types.go

```diff
@@ -27,7 +27,8 @@ type MemcachedSpec struct {
 }
 type MemcachedStatus struct {
 	// Nodes are the names of the memcached pods
-	Nodes []string `json:"nodes"`
+	// +kubebuilder:validation:UniqueItems=true
+	Nodes PodList `json:"nodes"`
```

Comment:
Mentioned this here: https://github.com/operator-framework/operator-sdk-samples/pull/82#discussion_r339078153

Curious if there's a new annotation we need to use? It seems like a structural schema requirement perhaps?

Reply:
The issue faced came from API rules validation of the open API lib used to generate the code and validations. The types for the validations will be more restrictive so the guess make 100% of sense for me as well.  But I still not sure if this change in the lib code is related to the structural schema or not yet.

Reply:
https://github.com/kubernetes/kube-openapi/issues/175
Confirmed. The []string should not work

Reply:
> Confirmed. The []string should not work

I'm not sure that's true. My reading of that issue and [this code](https://github.com/kubernetes/kube-openapi/blob/0270cf2f1c1d995d34b36019a6f65d58e6e33ad4/pkg/generators/rules/idl_tag.go#L27) is that it should not work _unless_ `// +listKind` is used on a comment for `Nodes []string`.

Can you try the following instead:
```go
type MemcachedStatus struct {
	// Nodes are the names of the memcached pods
	// +listKind
	Nodes PodList `json:"nodes"`
	...
}
```

Reply:
The ListType=set works :-) . Thank you for found it 👍